test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTypesTest.java

```diff
@@ -0,0 +1,1215 @@
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, map(2, 20), map(3, 30, 4, 40), 10)))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen user-defined types.
+     */
+    @Test
+    public void testNonFrozentuple()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a udt, b udt, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, {x:1, y:2}, {x:3, y:4}, 10)")
+                .queryColumns("a.x",
+                              rows(row(1)),
+                              rows(row(0, tuple(1, 2), tuple(3, 4), 10)),
+                              rows(flush ? row(0, tuple(1, 2), tuple(EMPTY_BYTE_BUFFER, EMPTY_BYTE_BUFFER), null)
+                                         : row(0, tuple(1, 2), tuple(3, 4), null))) // TODO: is this correct?
+                .deleteColumn("DELETE a.x FROM %s WHERE k=0",
+                              "b.y",
+                              rows(row(4)),
+                              rows(row(0, tuple(null, 2), tuple(3, 4), 10)),
+                              rows(row(0, tuple(null, 2), tuple(3, 4), 10))) // unselected 'a' is repaired, see CASSANDRA-10657
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, tuple(null, 2), tuple(3, 4), 10)))
+                .tearDown();
+    }
+
+    private Tester tester()
+    {
+        return new Tester(strategy, flush, localCoordinator, paging);
+    }
+
+    private static class Tester
+    {
+        private static final AtomicInteger seqNumber = new AtomicInteger();
+        private static final Object[][] EMPTY = new Object[0][];
+
+        private final String tableName = KEYSPACE + ".t_" + seqNumber.getAndIncrement();
+        private final ReadRepairStrategy strategy;
+        private final boolean flush;
+        private final boolean paging;
+        private final ICoordinator coordinator;
+
+        private String restriction; // the tested CQL query WHERE restriction
+        private String allColumnsQuery; // a SELECT * query for the table using the tested restriction
+
+        Tester(ReadRepairStrategy strategy, boolean flush, boolean localCoordinator, boolean paging)
+        {
+            this.strategy = strategy;
+            this.flush = flush;
+            this.paging = paging;
+
+            // decide which node we are going to use as coordinator for distributed reads, either the one containing the
+            // fresh data or the outdated one
+            coordinator = cluster.coordinator(localCoordinator ? 1 : 2);
+        }
+
+        Tester schemaChange(String... queries)
+        {
+            for (String query : queries)
+                cluster.schemaChange(withKeyspace(query));
+            return this;
```

Comment:
nit: Do we need to chain things w/ the `Tester`. Initially, I thought this might be something that would build a test specification, but it seems like it just runs the various stages as the chained methods are called.